src/commands/base-command.ts

```diff
@@ -551,6 +559,34 @@ export default class BaseCommand extends Command {
         process.env.NETLIFY_API_URL === `${apiUrl.protocol}//${apiUrl.host}` ? '/api/v1' : apiUrl.pathname
     }
 
+    const agent = await getAgent({
+      httpProxy: flags.httpProxy,
+      certificateFile: flags.httpProxyCertificateFilename,
+    })
+    const apiOpts = { ...apiUrlOpts, agent }
+    // TODO: remove typecast once we have proper types for the API
+    const api = new NetlifyAPI(token || '', apiOpts) as NetlifyOptions['api']
+
+    // If a user passes a site name as an option instead of a site ID to options.site, the siteInfo object
+    // will only have the property siteInfo.id. Checking for one of the other properties ensures that we can do
+    // a re-call of the api.getSite() that is done in @netlify/config so we have the proper site object in all
+    // commands.
+    // options.site as a site name (and not just site id) was introduced for the deploy command, so users could
+    // deploy by name along with by id
+    let siteIdByName = ''
+    if (flags.site) {
+      const site = await getSiteByName(api, flags.site)
```

Comment:
I might be wrong about what `getSiteByName` does, but isn't this going to lead to more than one request to the API to get the site given the logic on line 582?